sh/posix.bzl

```
@@ -181,11 +181,15 @@ MAKE_VARIABLES = "@rules_sh//sh/posix:make_variables"
 
 def _sh_posix_toolchain_impl(ctx):
     commands = {}
+    cmds = ctx.attr.cmds
     for cmd in _commands:
-        cmd_path = getattr(ctx.attr, cmd, None)
+        cmd_path = cmds.get(cmd, None)
         if not cmd_path:
             cmd_path = None
         commands[cmd] = cmd_path
+    unrecognizeds = [cmd for cmd in cmds.keys() if cmd not in _commands]
+    if unrecognizeds:
+        fail("Unrecognized commands in keys of sh_posix_toolchain's \"cmds\" attributes: {}. See _commands in posix.bzl for the list of recognized commands.".format(", ".join(unrecognizeds)))
```

```suggestion
        fail("Unrecognized commands in keys of sh_posix_toolchain's \"cmds\" attributes: {}. See posix.commands in @rules_sh//sh:posix.bzl for the list of recognized commands.".format(", ".join(unrecognizeds)))
```
Thanks for the informative error message. With `posix.commands` users can actually import that list, this updates the reference accordingly.

---

Change done in 59d5763